From e6c0dfac96246343b80d35bc698b0018f740554a Mon Sep 17 00:00:00 2001
Message-ID: <e6c0dfac96246343b80d35bc698b0018f740554a.1752669183.git.sam@gentoo.org>
In-Reply-To: <023bf5a7c304d113468217901401c80565240685.1752669183.git.sam@gentoo.org>
References: <023bf5a7c304d113468217901401c80565240685.1752669183.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Tue, 15 Jul 2025 04:50:21 +0100
Subject: [PATCH 3/4] gas: disable SFrame warnings

When bootstrapping Binutils in packaging, I saw:
```
{standard input}: Assembler messages:
{standard input}: Warning: no SFrame FDE emitted; non-SP/FP register 10 in .cfi_def_cfa
{standard input}: Warning: no SFrame FDE emitted; .cfi_escape with op (0xf)
[...]
FAIL: bootstrap with strip
```

Disable all the "no SFrame FDE emitted" warnings for now.

ChangeLog:

        * gas/gen-sframe.c (sframe_xlate_do_def_cfa): Disable warning.
        (sframe_xlate_do_def_cfa_register): Ditto.
        (sframe_xlate_do_def_cfa_offset): Ditto.
        (sframe_xlate_do_val_offset): Ditto.
        (sframe_xlate_do_register): Ditto.
        (sframe_xlate_do_remember_state): Ditto.
        (sframe_xlate_do_aarch64_negate_ra_state_with_pc): Ditto.
        (sframe_xlate_do_gnu_window_save): Ditto.
        (sframe_xlate_do_escape_expr): Ditto.
        (sframe_xlate_do_cfi_escape): Ditto.
        (sframe_xlate_do_cfi_undefined): Ditto.
        (sframe_do_cfi_insn): Ditto.
        (sframe_do_fde): Ditto.
---
 gas/gen-sframe.c | 29 -----------------------------
 1 file changed, 29 deletions(-)

diff --git a/gas/gen-sframe.c b/gas/gen-sframe.c
index 806ef5f9b5b..f2c4aa66ceb 100644
--- a/gas/gen-sframe.c
+++ b/gas/gen-sframe.c
@@ -1022,9 +1022,6 @@ sframe_xlate_do_def_cfa (struct sframe_xlate_ctx *xlate_ctx,
   if (cfi_insn->u.ri.reg != SFRAME_CFA_SP_REG
       && cfi_insn->u.ri.reg != SFRAME_CFA_FP_REG)
     {
-      as_warn (_("no SFrame FDE emitted; "
-		 "non-SP/FP register %u in .cfi_def_cfa"),
-	       cfi_insn->u.ri.reg);
       return SFRAME_XLATE_ERR_NOTREPRESENTED; /* Not represented.  */
     }
   sframe_fre_set_cfa_base_reg (cur_fre, cfi_insn->u.ri.reg);
@@ -1052,9 +1049,6 @@ sframe_xlate_do_def_cfa_register (struct sframe_xlate_ctx *xlate_ctx,
   if (cfi_insn->u.r != SFRAME_CFA_SP_REG
       && cfi_insn->u.r != SFRAME_CFA_FP_REG)
     {
-      as_warn (_("no SFrame FDE emitted; "
-		 "non-SP/FP register %u in .cfi_def_cfa_register"),
-	       cfi_insn->u.r);
       return SFRAME_XLATE_ERR_NOTREPRESENTED; /* Not represented.  */
     }
   sframe_fre_set_cfa_base_reg (cur_fre, cfi_insn->u.r);
@@ -1092,8 +1086,6 @@ sframe_xlate_do_def_cfa_offset (struct sframe_xlate_ctx *xlate_ctx,
     {
       /* No CFA base register in effect.  Non-SP/FP CFA base register should
 	 not occur, as sframe_xlate_do_def_cfa[_register] would detect this.  */
-      as_warn (_("no SFrame FDE emitted; "
-		 ".cfi_def_cfa_offset without CFA base register in effect"));
       return SFRAME_XLATE_ERR_NOTREPRESENTED;
     }
 
@@ -1157,9 +1149,6 @@ sframe_xlate_do_val_offset (const struct sframe_xlate_ctx *xlate_ctx ATTRIBUTE_U
 	      || (sframe_get_abi_arch () == SFRAME_ABI_S390X_ENDIAN_BIG
 		  && cfi_insn->u.ri.offset != SFRAME_S390X_SP_VAL_OFFSET))))
     {
-      as_warn (_("no SFrame FDE emitted; %s with %s reg %u"),
-	       cfi_esc_p ? ".cfi_escape DW_CFA_val_offset" : ".cfi_val_offset",
-	       sframe_register_name (cfi_insn->u.ri.reg), cfi_insn->u.ri.reg);
       return SFRAME_XLATE_ERR_NOTREPRESENTED; /* Not represented.  */
     }
 
@@ -1217,8 +1206,6 @@ sframe_xlate_do_register (struct sframe_xlate_ctx *xlate_ctx ATTRIBUTE_UNUSED,
       /* Ignore SP reg, as it can be recovered from the CFA tracking info.  */
       )
     {
-      as_warn (_("no SFrame FDE emitted; %s register %u in .cfi_register"),
-	       sframe_register_name (cfi_insn->u.rr.reg1), cfi_insn->u.rr.reg1);
       return SFRAME_XLATE_ERR_NOTREPRESENTED;  /* Not represented.  */
     }
 
@@ -1239,8 +1226,6 @@ sframe_xlate_do_remember_state (struct sframe_xlate_ctx *xlate_ctx)
      info for the function involved.  */
   if (!last_fre)
     {
-      as_warn (_("no SFrame FDE emitted; "
-		 ".cfi_remember_state without prior SFrame FRE state"));
       return SFRAME_XLATE_ERR_INVAL;
     }
 
@@ -1333,7 +1318,6 @@ static int
 sframe_xlate_do_aarch64_negate_ra_state_with_pc (struct sframe_xlate_ctx *xlate_ctx ATTRIBUTE_UNUSED,
 						 struct cfi_insn_data *cfi_insn ATTRIBUTE_UNUSED)
 {
-  as_warn (_("no SFrame FDE emitted; .cfi_negate_ra_state_with_pc"));
   /* The used signing method should be encoded inside the FDE in SFrame v3.
      For now, PAuth_LR extension is not supported with SFrame.  */
   return SFRAME_XLATE_ERR_NOTREPRESENTED;  /* Not represented.  */
@@ -1360,7 +1344,6 @@ sframe_xlate_do_gnu_window_save (struct sframe_xlate_ctx *xlate_ctx,
       || abi_arch == SFRAME_ABI_AARCH64_ENDIAN_LITTLE)
     return sframe_xlate_do_aarch64_negate_ra_state (xlate_ctx, cfi_insn);
 
-  as_warn (_("no SFrame FDE emitted; .cfi_window_save"));
   return SFRAME_XLATE_ERR_NOTREPRESENTED;  /* Not represented.  */
 }
 
@@ -1418,9 +1401,6 @@ sframe_xlate_do_escape_expr (const struct sframe_xlate_ctx *xlate_ctx,
       || (sframe_ra_tracking_p () && reg == SFRAME_CFA_RA_REG)
       || reg == xlate_ctx->cur_fre->cfa_base_reg)
     {
-      as_warn (_("no SFrame FDE emitted; "
-		 ".cfi_escape DW_CFA_expression with %s reg %u"),
-	       sframe_register_name (reg), reg);
       err = SFRAME_XLATE_ERR_NOTREPRESENTED;
     }
   /* else safe to skip, so continue to return SFRAME_XLATE_OK.  */
@@ -1579,8 +1559,6 @@ sframe_xlate_do_cfi_escape (const struct sframe_xlate_ctx *xlate_ctx,
 	 OS-specific CFI opcodes), skip inspecting the DWARF expression.
 	 This may impact the asynchronicity due to loss of coverage.
 	 Continue to warn the user and bail out.  */
-      as_warn (_("no SFrame FDE emitted; .cfi_escape with op (%#lx)"),
-	       (unsigned long)firstop);
       err = SFRAME_XLATE_ERR_NOTREPRESENTED;
     }
 
@@ -1604,8 +1582,6 @@ sframe_xlate_do_cfi_undefined (const struct sframe_xlate_ctx *xlate_ctx ATTRIBUT
       || cfi_insn->u.r == SFRAME_CFA_RA_REG
       || cfi_insn->u.r == SFRAME_CFA_SP_REG)
     {
-      as_warn (_("no SFrame FDE emitted; %s reg %u in .cfi_undefined"),
-	       sframe_register_name (cfi_insn->u.r), cfi_insn->u.r);
       return SFRAME_XLATE_ERR_NOTREPRESENTED; /* Not represented.  */
     }
 
@@ -1782,8 +1758,6 @@ sframe_do_cfi_insn (struct sframe_xlate_ctx *xlate_ctx,
 
 	if (!cfi_name)
 	  cfi_name = _("(unknown)");
-	as_warn (_("no SFrame FDE emitted; CFI insn %s (%#x)"),
-		 cfi_name, op);
 	err = SFRAME_XLATE_ERR_NOTREPRESENTED;
       }
     }
@@ -1806,8 +1780,6 @@ sframe_do_fde (struct sframe_xlate_ctx *xlate_ctx,
   /* SFrame format cannot represent a non-default DWARF return column reg.  */
   if (xlate_ctx->dw_fde->return_column != DWARF2_DEFAULT_RETURN_COLUMN)
     {
-      as_warn (_("no SFrame FDE emitted; non-default RA register %u"),
-	       xlate_ctx->dw_fde->return_column);
       return SFRAME_XLATE_ERR_NOTREPRESENTED;
     }
 
@@ -1851,7 +1823,6 @@ sframe_do_fde (struct sframe_xlate_ctx *xlate_ctx,
 	  if (fre->ra_loc != SFRAME_FRE_ELEM_LOC_STACK
 	      && fre->bp_loc == SFRAME_FRE_ELEM_LOC_STACK)
 	    {
-	      as_warn (_("no SFrame FDE emitted; FP without RA on stack"));
 	      return SFRAME_XLATE_ERR_NOTREPRESENTED;
 	    }
 	}
-- 
2.50.1

